apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java

```diff
@@ -266,29 +272,181 @@ void testDeactivationBeforeEnd() throws Exception {
+            "1 2    2 3 3 1"
         }, new Object[][] {
-            {"a",       9},
+            {"a",       8},
             {"  b",     5},
         }, new Object[][] {
-            {"1",       14},
-            {"  a",     12},
+            {"1",       13},
+            {"  a",     11},
             {"    b",    6},
             {"      2",  5},
             {"    3",    2},
         });
     }
 
+    /*
+     * [1       ]
+     *  [a     ]
+     *   [b][2]
+     */
+    @Test
+    void testActivationAfterMethodEnds() throws Exception {
+        assertCallTree(new String[]{
+            " bbb      ",
+            " aaa aa a ",
+            "1   2  2 1"
+        }, new Object[][] {
+            {"a",       6},
+            {"  b",     3},
+        }, new Object[][] {
+            {"1",       9},
+            {"  a",     7},
+            {"    b",   2},
+            {"    2",   3}
+        });
+    }
+
+    /*
+     * [1         ]
+     *  [a][2    ]
+     *  [b] [3  ]
+     *       [c]
+     */
+    @Test
+    void testNestedActivationAfterMethodEnds() throws Exception {
+        Map<String, AbstractSpan<?>> spans = assertCallTree(new String[]{
+            " bbb        ",
+            " aaa  ccc   ",
+            "1   23   321"
+        }, new Object[][] {
+            {"a",        3},
+            {"  b",      3},
+            {"c",        3},
+        }, new Object[][] {
+            {"1",       11},
+            {"  b",      2, List.of("a")},
+            {"  2",      6},
+            {"    3",    4},
+            {"      c",  2}
+        });
+        assertThat(spans.get("b").getChildIds().getSize()).isEqualTo(0);
+    }
+
+    /*
+     * [1           ]
+     *  [a  ][3    ]
+     *  [b  ] [4  ]
+     *   [2]   [c]
+     */
+    @Test
+    void testRegularActivationFollowedByNestedActivationAfterMethodEnds() throws Exception {
+        assertCallTree(new String[]{
+            "   d          ",
+            " b b b        ",
+            " a a a  ccc   ",
+            "1 2 2 34   431"
+        }, new Object[][] {
+            {"a",        3},
+            {"  b",      3},
+            {"c",        3},
+        }, new Object[][] {
+            {"1",       13},
+            {"  b",      4, List.of("a")},
+            {"    2",    2},
+            {"  3",      6},
+            {"    4",    4},
+            {"      c",  2}
+        });
+    }
+
+    /*
+     * [1           ]
+     *  [a         ]
+     *   [b][2    ]
+     *       [3  ]
+     *        [c]
+     */
+    @Test
+    void testNestedActivationAfterMethodEnds2() throws Exception {
+        Map<String, AbstractSpan<?>> spans = assertCallTree(new String[]{
+            "  bbb  ccc    ",
+            " aaaa  aaa  a ",
+            "1    23   32 1"
+        }, new Object[][]{
+            {"a", 8},
+            {"  b", 3},
+            {"  c", 3},
+        }, new Object[][]{
+            {"1", 13},
+            {"  a", 11},
+            {"    b", 2},
+            {"  2", 6},
```

Comment:
How come `2` is not a child of `a`?

Reply:
Wow, good catch! From now on, I call you the eagle eye. Or maybe Eyeal 🤔 

Submitting fix shortly that searches for common ancestors, instead of always adding the child_ids on the top of the stack